### ISSUE TITLE
Filter Admins via remote_user

### ISSUE THREAD

**Issue by jberanek** [meeting-room-booking-system/mrbs-code#2402]
We are trying to implement our SSO with MRBS and are using the Remote_User authentication scheme. 

However, I cannot seem to distinguish the users between Admin and normal user. Is there a way to do this when using a remote_user authentication scheme? 

Christopher Schneider
UChicago - Polsky Center

Reported by: *anonymous

Original Ticket: [mrbs/support-requests/1700](https://sourceforge.net/p/mrbs/support-requests/1700)

**Comment by jonkloske** [comment]
I had what I thought was the same issue here, but in my case I was trying to be clever: because the doco says we should expect $_SERVER['REMOTE_USER'] to be used, I set that inside config.inc.php based off the slightly different $_SERVER variable that holds our locally logged in username.

After a lot of poking around in the codebase, it turns out a few includes before config.inc.php there's grab_globals.inc.php which snarfs the entire $_SERVER array into a $server global var, and then everything else uses that from then on.

After having config.inc.php update the $server['REMOTE_USER'] var instead, it was a lot happier.

I also had a similar issue with the log off link, which turns out was related to the fact I was (unknowingly) not logged in at all!

**Comment by campbell-m** [comment]
> After a lot of poking around in the codebase, it turns out a few includes before config.inc.php there's grab_globals.inc.php which snarfs the entire $_SERVER array into a $server global var, and then everything else uses that from then on.

Yes, this is to get round problems when authenticating against WordPress, which automatically slash-escapes (or certainly used to - I haven't checked whether it still does) the superglobal values.  MRBS grabs a copy of the variables before WordPress has a chance to tamper with them.
